app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeightForm.java

```diff
@@ -0,0 +1,196 @@
+package de.westnordost.streetcomplete.quests.max_height;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.quests.AbstractQuestFormAnswerFragment;
+import de.westnordost.streetcomplete.view.dialogs.AlertDialogBuilder;
+
+public class AddMaxHeightForm extends AbstractQuestFormAnswerFragment
+{
+	public static final String
+		MAX_HEIGHT = "max_height",
+		NO_SIGN = "no_sign";
+
+	private EditText heightInput, feetInput, inchInput;
+
+	private boolean isMetric;
+
+	@Override public View onCreateView(LayoutInflater inflater, ViewGroup container,
+									   Bundle savedInstanceState)
+	{
+		View view = super.onCreateView(inflater, container, savedInstanceState);
+
+		isMetric = getCountryInfo().getHeightUnits().get(0).equals("m");
+
+		setMaxHeightSignLayout(getMaxHeightLayoutResourceId());
+		addOtherAnswers();
+
+		return view;
+	}
+
+	private void setMaxHeightSignLayout(int resourceId)
+	{
+		View contentView = setContentView(resourceId);
+
+		heightInput = contentView.findViewById(R.id.maxHeightInput);
+		feetInput = contentView.findViewById(R.id.maxHeightFeetInput);
+		inchInput = contentView.findViewById(R.id.maxHeightInchInput);
+	}
+
+	private int getMaxHeightLayoutResourceId()
+	{
+		return isMetric ? R.layout.quest_max_height : R.layout.quest_max_height_us;
```

Comment:
Unfortunately, the US is not the only country that uses imperial units. You need to separate the selection of the layout from the selection of which unit to select. [I.e. this is how the signs look like in UK apparently.](https://en.wikipedia.org/wiki/File:UK_traffic_sign_629.2A.svg)

1. Traffic sign layout: most countries follow the [International Standard](https://en.wikipedia.org/wiki/Vienna_Convention_on_Road_Signs_and_Signals), see also the [comparison of European road signs](https://en.wikipedia.org/wiki/Comparison_of_European_road_signs). This is the layout I saw in your screenshot. However, I think you should also include the "**m**" in the form, even if it is not a dropdown menu, because it is an element in **all** the signs.
Some countries follow the [US standard (MUTCD)](https://en.wikipedia.org/wiki/Comparison_of_MUTCD-influenced_traffic_signs). In the linked page, search for "Height restriction ahead" to see how the sign look there. So, the layout should look [like that](https://en.wikipedia.org/wiki/Comparison_of_MUTCD-influenced_traffic_signs#/media/File:Brasil_A-37.svg). If you feel extra-cool, you could make extra forms even on variations, i.e. for Australia.

2. Meters vs Feet+Inch: So, both layouts need to have both input fields. These input fields can simply be made visible/invisible depending on the country or user's selection where both systems may be used.